### ISSUE TITLE
iptables error in basic-install.sh

### ISSUE THREAD

**Issue by WaLLy3K** [pi-hole/pi-hole#1125]
**In raising this issue, I confirm the following:**

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*
- [x] The issue I'm reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the codebase?:**

4

---
**[BUG] Expected Behaviour:**
The automated installation script should not throw any errors when installing on a Raspberry Pi 3 on Debian Jesse Lite (Specifically, DietPi v141)

**[BUG] Actual Behaviour:**
````
::: Installing latest Cron script... done!
bash: line 894: iptables: command not found
bash: line 894: iptables: command not found
:::
::: Preparing to run gravity.sh to refresh hosts...
````

**[BUG] Steps to reproduce:**
Run `curl -sSL https://install.pi-hole.net | bash` on a Raspberry Pi without going out of your way to install iptables.

If I understand correctly: if a user installs iptables, they will have a command they can use that's just called `iptables`. Using `$(which iptables)` wrapped in an IF statement should be able to test for a valid installation rather than [just checking if `ip_tables.ko` is installed](https://github.com/pi-hole/pi-hole/blob/master/automated%20install/basic-install.sh#L891).

**Comment by DL6ER** [comment]
I did exactly what you did today (using `2017-01-11-raspbian-jessie-lite.img`) and have seen no issue! The first thing I did was running `curl -sSL https://install.pi-hole.net | bash` after logging in for the first time.

I had `iptables` installed from the beginning using this image. Did you change something?

**Comment by WaLLy3K** [comment]
Shoot, I was about to make a sneaky edit! I'm specifically running [DietPi v141](http://dietpi.com) (latest), which is possible that it may not use the absolute latest version Rasbian Jesse yet.

Other than that, I've made no specific changes in my install for iptables.

**Comment by DL6ER** [comment]
Okay, `iptables` is too heavy for DietPi which in not like Debian Jesse Lite anymore. These messages won't affect your installation since nothing was done (as `iptables` was not around). We will silence the messages you have been seeing.

**Comment by dschaper** [comment]
If DietPi ships without a firewall solution of some type, then I'd be wary of installing server daemons on it. Do you know what package FourD uses for his firewall? Debian by default comes with IPTables (as does just about any linux distribution.)

**Comment by WaLLy3K** [comment]
@dschaper: Honestly, don't know. I've asked via [Twitter](https://twitter.com/WaLLy3K/status/819307585434042372) though.

**Comment by PromoFaux** [comment]
We can ask him here too..! @Fourdee what say you?

**Comment by Fourdee** [comment]
> We can ask him here too..! @Fourdee what say you?

@PromoFaux I have been summoned lol 😃 

> If DietPi ships without a firewall solution of some type, then I'd be wary of installing server daemons on it. Do you know what package FourD uses for his firewall? Debian by default comes with IPTables (as does just about any linux distribution.)

With DietPi, `iptables` is installed on demand, as needed to allow for specific network routing (eg: hostapd, hotspot)

As for firewall, 99.9% of users routers have a built in firewall, which will protect your internal network. Effectively, your router does that job and another firewall is unnecessary on client systems in the local network. 

Regardless, if you believe you "need" a firewall on your system, you can install one via ```apt```.

As for this ticket, i'll add ```iptables``` to the prereqs in DietPi for PiHole installation. Whether you guys want to add ```apt-get install iptables``` to your ```basic-install.sh``` is your call, as this only really affects DietPi image due to it being minimal.

**Comment by PromoFaux** [comment]
I'm not sure there is any need for you to install it, especially if you've left it out for any reason. 

But obviously we're detecting it as the kernel-module is still there, just not the command. 

Take a look at [the relevant line in the install script](https://github.com/pi-hole/pi-hole/blob/master/automated%20install/basic-install.sh#L891), If we don't find it, we don't bother setting the rules.

@WaLLy3K suggested we replace it with an `if $(which iptables)`, and I think that is the correct way to go about it. I was just wondering if it was more an accidental oversight on your part as regards the `iptables` command.

**Comment by Fourdee** [comment]
> @WaLLy3K suggested we replace it with an if $(which iptables), and I think that is the correct way to go 

That would do it 👍.

**Comment by PromoFaux** [comment]
I'd say you can safely close https://github.com/Fourdee/DietPi/issues/708 then :)

Unless you specifically want to add it in, that is!

**Comment by Fourdee** [comment]
> I'd say you can safely close Fourdee/DietPi#708 then :)
Unless you specifically want to add it in, that is!

Yep, as ```iptables``` is not required, we'll leave it out.

**Comment by dschaper** [comment]
Fixed in #1136
